test/ip-addresses.ts

```diff
@@ -20,86 +20,132 @@ import {
 } from '../src/ip-addresses';
 
 t.throws(
-  () => parseIpAddresses(undefined),
+  () => parseIpAddresses(undefined, undefined),
   {code: 'ENOSQLADMINIPADDRESS'},
   'should throw if no argument'
 );
 
 t.throws(
-  () => parseIpAddresses([]),
+  () => parseIpAddresses([], undefined),
   {code: 'ENOSQLADMINIPADDRESS'},
   'should throw if no ip is found'
 );
 
 t.same(
-  parseIpAddresses([
-    {
-      ipAddress: '0.0.0.0',
-      type: 'PRIMARY',
-    },
-  ]),
+  parseIpAddresses(
+    [
+      {
+        ipAddress: '0.0.0.0',
+        type: 'PRIMARY',
+      },
+    ],
+    undefined
+  ),
   {
     public: '0.0.0.0',
   },
   'should return a public ip successfully'
 );
 
 t.same(
-  parseIpAddresses([
-    {
-      ipAddress: '0.0.0.0',
-      type: 'PRIMARY',
-    },
-    {
-      ipAddress: '0.0.0.1',
-      type: 'OUTGOING',
-    },
-  ]),
+  parseIpAddresses(
+    [
+      {
+        ipAddress: '0.0.0.0',
+        type: 'PRIMARY',
+      },
+      {
+        ipAddress: '0.0.0.1',
+        type: 'OUTGOING',
+      },
+    ],
+    undefined
+  ),
   {
     public: '0.0.0.0',
   },
   'should return a public ip from a list'
 );
 
 t.same(
-  parseIpAddresses([
-    {
-      ipAddress: '0.0.0.2',
-      type: 'PRIVATE',
-    },
-    {
-      ipAddress: '0.0.0.1',
-      type: 'OUTGOING',
-    },
-  ]),
+  parseIpAddresses(
+    [
+      {
+        ipAddress: '0.0.0.2',
+        type: 'PRIVATE',
+      },
+      {
+        ipAddress: '0.0.0.1',
+        type: 'OUTGOING',
+      },
+    ],
+    undefined
+  ),
   {
     private: '0.0.0.2',
   },
   'should return a private ip from a list'
 );
 
 t.same(
-  parseIpAddresses([
-    {
-      ipAddress: '0.0.0.0',
-      type: 'PRIMARY',
-    },
-    {
-      ipAddress: '0.0.0.2',
-      type: 'PRIVATE',
-    },
-    {
-      ipAddress: '0.0.0.1',
-      type: 'OUTGOING',
-    },
-  ]),
+  parseIpAddresses(
+    [
+      {
+        ipAddress: '0.0.0.0',
+        type: 'PRIMARY',
+      },
+      {
+        ipAddress: '0.0.0.2',
+        type: 'PRIVATE',
+      },
+      {
+        ipAddress: '0.0.0.1',
+        type: 'OUTGOING',
+      },
+    ],
+    undefined
+  ),
   {
     private: '0.0.0.2',
     public: '0.0.0.0',
   },
   'should return a both public and private ips if available'
 );
 
+t.same(
+  parseIpAddresses([], '0.0.0.3'),
```

Comment:
can we use `'abcde.12345.us-central1.sql.goog'` as the test `dnsName` variable? This is the format the dnsName takes in an actual response.